dependencies/test.py

```diff
@@ -0,0 +1,146 @@
+import os
+import pickle
+
+import requests
+from os.path import join, getsize
+import lxml.html
+
+from node import node
+
+
+class ReadFiles:
+    path=""
+
+    def __init__(self, path):
+        self.path = path
+
+    def getAllFiles(self):
+        list = []
+        for root, dirs, files in os.walk(self.path):
+            for name in files:
+                if name.endswith(".spec"):
+                    list.append(name)
```

Comment:
what will happen if you'll have 1M entries in a list? Instead better use generators